packages/trader/src/Modules/Contract/Containers/contract-replay.tsx

```diff
@@ -0,0 +1,384 @@
+    const { contract_store } = contract_replay;
+    const {
+        is_market_closed,
+        is_sell_requested,
+        onClickCancel,
+        onClickSell,
+        onMount,
+        onUnmount,
+        indicative_status,
+        is_chart_loading,
+        is_forward_starting,
+    } = contract_replay;
+    const { contract_info, contract_update, contract_update_history, is_digit_contract } = contract_store;
+    const { routeBackInApp } = common;
+    const { is_dark_mode_on: is_dark_theme, notification_messages_ui: NotificationMessages, toggleHistoryTab } = ui;
+    const trade_type_feature_flag =
+        contract_info.shortcode &&
+        getContractTypeFeatureFlag(contract_info.contract_type ?? '', isHighLow(contract_info));
+    const is_trade_type_disabled =
+        useFeatureFlags()[`is_${trade_type_feature_flag}_enabled` as keyof ReturnType<typeof useFeatureFlags>] ===
+        false;
+    const [is_visible, setIsVisible] = React.useState(false);
+    const history = useHistory();
+
+    React.useEffect(() => {
+        const url_array = /[^/]*$/.exec(location.pathname);
+        const url_contract_id = url_array ? +url_array[0] : undefined;
+        onMount(contract_id || url_contract_id);
+        setIsVisible(true);
+
+        return () => {
+            setIsVisible(false);
+            onUnmount();
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [contract_id, location, onMount, onUnmount]);
+
+    const onClickClose = React.useCallback(() => {
+        setIsVisible(false);
+        const is_from_table_row = !isEmptyObject((location as TLocation).state)
+            ? (location as TLocation).state.from_table_row
+            : false;
+        return is_from_table_row ? history.goBack() : routeBackInApp(history);
+    }, [history, routeBackInApp]);
+
+    React.useEffect(() => {
+        // don't open Contract details page for trade types with disabled feature flag:
+        if (is_trade_type_disabled && is_visible) {
+            onClickClose();
+        }
+    }, [is_trade_type_disabled, is_visible, onClickClose]);
+
+    const onChangeSwipeableIndex = (index: number) => {
+        setSwipeIndex(index);
+    };
+
+    if (!contract_info.underlying) return null;
+
+    const is_accumulator = isAccumulatorContract(contract_info.contract_type);
+    const is_multiplier = isMultiplierContract(contract_info.contract_type);
+    const is_turbos = isTurbosContract(contract_info.contract_type);
+    const is_vanilla = isVanillaContract(contract_info.contract_type);
+    const is_smarttrader_contract = isSmartTraderContract(contract_info.contract_type);
+
+    const contract_drawer_el = (
+        <ContractDrawer
+            contract_info={contract_info}
+            contract_update={contract_update}
+            contract_update_history={contract_update_history}
+            is_accumulator={is_accumulator}
+            // is_chart_loading={is_chart_loading}
```

Comment:
I temporary comment this line as `<ContractDrawer />` do not expect this prop. Is it needed for triggering re-render? Need to check